geopolars/src/geodataframe.rs

```diff
@@ -3,11 +3,17 @@ use polars::prelude::{DataFrame, Result, Series};
 
 pub trait GeoDataFrame {
     fn centroid(&self) -> Result<Series>;
+    fn convex_hull(&self) -> Result<Series>;
 }
 
 impl GeoDataFrame for DataFrame {
     fn centroid(&self) -> Result<Series> {
         let geom_column = self.column("geometry")?;
         geom_column.centroid()
     }
+
+    fn convex_hull(&self) -> Result<Series> {
```

Comment:
I'm not sure what the right approach is here. This `GeoDataFrame` construct was just hacking; you can see the hard coded `"geometry"`. Ideally we'd be able to store dataframe-level metadata about a "primary geometry column" as I think geopandas does.

Reply:
Humm, trying to think what functionality is available on a GeoPandas dataframe that isn't just calling down to the GeoSeries .

Some of the plotting functions for sure, where you need access to both the geometry column and a variable to color the geometries with. Some of the spatial join functions might naturally live at the DataFrame level as well?

I like the idea of having a primary geometry column like GeoPandas.

Reply:
It seems like _most_ functions don't _need_ a primary geo column set. Like in pandas you have `pd.merge(left, right, left_on, right_on)`, and so you could do the same with a spatial join where you just define which column in each df you should merge on.

I'm not necessarily opposed to having a primary geo column but it seems like it's blocked anyways on polars exposing dataframe-level metadata (unless we just declare the primary geo column needs to have the name "geometry").

Reply:
Yeah, I think that it's handy for convenience or when the DataFrame only has a single geometry column. 

Sorry the join example was more in thinking about the usefulness of GeoDataFrame construct in general rather than the idea of a primary geometry column.

Reply:
This is fine for now and we can come back to what the geodataframe API should look like